src/main/java/game/map/Map.java

```diff
@@ -76,5 +81,42 @@ public List<Country> getCountries() {
     public List<Player> getPlayers() {
         return players;
     }
+    public static void assignCountries(List<Player> players, List<Country> countries) {
+
+                HashMap<Player, List<Country>> playerCountryMap = new HashMap<>();
+
+                if (countries.isEmpty()) {
+                    System.out.println("No countries available to assign.");
+
+                }
+                Collections.shuffle(countries);
+
+                int numCountriesPerPlayer = (countries.size()) / (players.size());
+
+                int currentIndex = 0;
+                for (Player player : players) {
+                    List<Country> assignedCountries = new ArrayList<>();
+                    for (int i = 0; i < numCountriesPerPlayer; i++) {
+                        assignedCountries.add(countries.get(currentIndex));
+                        currentIndex++;
+                    }
+                    player.setAssignedCountries(assignedCountries);
```

Comment:
this can create issues, because you are assigning a local variable here to an object's property. Instead I would suggest use the below,
`player.getCountries().addAll(assignedCountries);`